src/Symfony/Component/Finder/Iterator/FilenameFilterIterator.php

```diff
@@ -42,6 +51,6 @@ public function accept()
      */
     protected function toRegex($str)
     {
-        return $this->isRegex($str) ? $str : Glob::toRegex($str);
+        return $this->isRegex($str) ? $str.($this->isCaseSensitive ? '' : 'i') : Glob::toRegex($str, true, true, '#', $this->isCaseSensitive);
```

Comment:
What if we already received the 'i' modifier from the i/name method?
Shouldn't that be checked here to avoid the duplicate modifier?